File: JAVAC SCRIPT QUESTION/atmCard.js
function atmCard(cardNumber) {

    let issuer = "Unknown";
    let isValid = false;
    const count = cardNumber.length;

    
    if (cardNumber[0] === 5 && count === 16) {
        issuer = "Mastercard";
    } else if (cardNumber[0] === 6 && count === 16) {
        issuer = "Discover";
    } else if (cardNumber[0] === 3 && count === 16) {
        issuer = "America Express";
    } else if (cardNumber[0] === 4 && count === 16) {
        issuer = "Visa";
    }

    
    let sum1 = 0;
    let sum2 = 0;

    
    for (let count = cardNumber.length - 2; count >= 0; count -= 2) {
        let product = cardNumber[count] * 2;
        if (product > 9) {
            product = Math.floor(product / 10) + (product % 10);
        }
        sum1 += product;
    }

    
    for (let count = cardNumber.length - 1; count >= 0; count -= 2) {
        sum2 += cardNumber[count];
    }

    const total = sum1 + sum2;
    isValid = (total % 10 === 0);

    
    console.log("Issuer:", issuer);
    console.log("Valid:", isValid);
}


const cardNumber = [5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 4, 4, 4, 4];
atmCard(cardNumber);

File: PYTHON QUESTION/atm_card.py
def atm_card(card_number):
    issuer = "Unknown"
    is_valid = False

    
    count = 0
    for _ in card_number:
        count += 1

    
    if card_number[0] == 5 and count == 16:
        issuer = "Mastercard"
    elif card_number[0] == 6 and count == 16:
        issuer = "Discover"
    elif card_number[0] == 3 and count == 16:
        issuer = "America Express"
    elif card_number[0] == 4 and count == 16:
        issuer = "Visa"

    sum1 = 0
    sum2 = 0

   
    index = count - 2
    while index >= 0:
        product = card_number[index] * 2
        if product > 9:
            product = product // 10 + product % 10
        sum1 += product
        index -= 2

    index = count - 1
    while index >= 0:
        sum2 += card_number[index]
        index -= 2

    total = sum1 + sum2
    is_valid = (total % 10 == 0)

    print("Issuer:", issuer)
    print("Valid:", is_valid)



card_number = [5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 4, 4, 4, 4]
atm_card(card_number)
